src/bio/bio_monitor.c

```diff
@@ -0,0 +1,541 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B620873.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+#define D_LOGFAC	DD_FAC(bio)
+
+#include <spdk/nvme.h>
+#include <spdk/bdev.h>
+#include <spdk/io_channel.h>
+#include "bio_internal.h"
+#include <daos_srv/smd.h>
+
+/* Period to query SPDK device health stats (1 min period) */
+#define DAOS_SPDK_STATS_PERIOD	(60 * (NSEC_PER_SEC / NSEC_PER_USEC))
+/* Used to preallocate buffer to query error log pages from SPDK health info */
+#define DAOS_MAX_ERROR_LOG_PAGES 256
+
+uint64_t io_stat_period;
```

Comment:
not necessary to have this here.  It should be covered by the extern in the header file.